app-backend/api/src/main/scala/shape/Routes.scala

```diff
@@ -44,12 +47,63 @@ trait ShapeRoutes extends Authentication
       get { listShapes } ~
       post { createShape }
     } ~
-    pathPrefix(JavaUUID) { shapeId =>
-      pathEndOrSingleSlash {
-        get { getShape(shapeId) } ~
-        put { updateShape(shapeId) } ~
-        delete { deleteShape(shapeId) }
+      post {
+        pathPrefix("upload") {
+          pathEndOrSingleSlash {
+            authenticate { user =>
+              val tempFile = ScalaFile.newTemporaryFile()
+              val response = storeUploadedFile("name", (_) => tempFile.toJava) { (m, _) =>
+                processShapefile(user, tempFile, m)
+              }
+              tempFile.delete()
+              response
+            }
+          }
+        } ~
+          pathPrefix(JavaUUID) { shapeId =>
+            pathEndOrSingleSlash {
+              get { getShape(shapeId) } ~
+                put { updateShape(shapeId) } ~
+                delete { deleteShape(shapeId) }
+            }
+          }
+      }
+  }
+
+  def processShapefile(user: User, tempFile: ScalaFile, fileMetadata: FileInfo) = {
+    val unzipped = tempFile.unzip()
+    val matches = unzipped.glob("*.shp")
+    matches.hasNext match {
+      case true => {
+        val shapeFile = matches.next()
+        val features = ShapeFileReader.readMultiPolygonFeatures(shapeFile.toString)
+
+        // Only reads first feature
+        features match {
+          case Nil => complete(StatusCodes.ClientError(400)("Bad Request", "No MultiPolygons detected in Shapefile"))
+          case feature +: _ => {
+            val geometry = feature.geom
+            val reprojectedGeometry = Projected(Reproject(geometry, LatLng, WebMercator), 3857)
+            reprojectedGeometry.isValid match {
+              case true => {
+                val shape = Shape.create(
+                  Some(user.id),
+                  user.organizationId,
+                  fileMetadata.fileName,
+                  None,
+                  Some(reprojectedGeometry)
+                )
+                complete(Shapes.insertShapes(Seq(shape), user))
```

Comment:
does this actually return a 201? I think it's 200 by default

Reply:
yup returns 200. update this or the swagger spec
![image](https://user-images.githubusercontent.com/4392704/35297800-1e44fd30-004e-11e8-89c0-b56fcc35b3c8.png)

Reply:
Fixed in https://github.com/raster-foundry/raster-foundry/pull/2919/commits/bf24be85842e4850bf29c0c9625fd8fe9984c41f